### ISSUE TITLE
System edit privileges required to permit non-admin to change camera mode

### ISSUE THREAD

**Issue by blacksail29** [ZoneMinder/ZoneMinder#1994]
If i use one other user that admin i must give this user the system EDIT privileges to permit to him to change the mode of his camera: ie from MODECT to MONITOR for example.

And of course, if i give this user the system/EDIT privs he can change the mode of all the cameras (global config in your app) and stop or restart the ZM server.

Without this edit priv, the user can't change the mode from ZmNinja.

I've tested this on the console and with ZM VIEW, it works fine. So it doesn't seem to be a user account privileges or trouble but perhaps one Zmninja/API one ?

Zmninja author replyed this :+1: 
This is a ZoneMinder API issue - you might want to open an issue in ZM
The code that is responsible for this is https://github.com/ZoneMinder/ZoneMinder/blob/master/web/api/app/Controller/MonitorsController.php#L125

Basically ,the monitor API allows you to edit many monitor parameters why is why its currently in System Edit permission.

For your purposes, you can try changing web/api/app/Controller/MonitorsController.php line 15 to if ($this->Session->Read('monitorPermission') != 'Edit') and see if it works. If this is how ZM console behaves, you might want to create an issue in ZoneMinder with this fix.

zmView doesn't use the API, so the API layer permissions don't apply to it.

Can this be patched in ZM files ?

thanks

eric
